Support for DSC Alarm Motion Sensors

Monstergerm opened the pull request:
Added support for Motion Sensors from DSC plugin.
Added some logic to verify correct number entries in device setup. Updated support URL to point to correct Wiki. Fixed rare python error when no device was selected in device setup.

Monstergerm committed:
Support for DSC Alarm Motion Sensors

Added support for Motion Sensors from DSC plugin.
Added some logic to verify correct number entries in device setup. Updated support URL to point to correct Wiki. Fixed rare python error when no device was selected in device setup.

FlyingDiver commented:
I fixed the Support  URL in an update I just pushed.  The support for DSC alarms is unlikely to be merged.  Using "model" is a problem as it will break for other alarm systems (or any plugin) that use the same "model" but with different state values.  The Masquerade plugin was specifically designed to do exactly this, and I hate replicating functionality into multiple plugins.  I do need to look at the other changes you made, but that won't be for another week or so.

Monstergerm commented:
Hi,
I totally understand about not wanting to replicate functionality. For me it was a small and simple change to add to the Occupatum plugin, which now makes it easy to support motion zones from the DSC plugin. Regarding the use of “model”, I added another bit of code to also confirm that “alarmZone” is coming from the DSC plugin. So it will not get mixed up with other plugins using the same model.

Here is my latest version with these changes. And safe travel ...



> On May 12, 2022, at 4:31 AM, Joe Keenan ***@***.***> wrote:
> 
> 
> I fixed the Support URL in an update I just pushed. The support for DSC alarms is unlikely to be merged. Using "model" is a problem as it will break for other alarm systems (or any plugin) that use the same "model" but with different state values. The Masquerade plugin was specifically designed to do exactly this, and I hate replicating functionality into multiple plugins. I do need to look at the other changes you made, but that won't be for another week or so.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/FlyingDiver/Indigo-Occupatum/pull/7#issuecomment-1124879565>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFA34VIA2G3IIJ7H3IX3BGDVJTTZPANCNFSM5VWVUY3Q>.
> You are receiving this because you authored the thread.
>

FlyingDiver commented:
I merged in all the changes not pertaining to the DSC devices.  If you want to keep it in your version, fine with me.
